distribution/templates/deploy.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+from __future__ import print_function
+from xml.etree import ElementTree
+
+import hashlib
+import os
+import re
+import subprocess as sp
+import sys
+import tempfile
+import zipfile
+from posixpath import join as urljoin
+
+# usual importing is not possible because
+# this script and module with common functions
+# are at different directory levels in sandbox
+from runpy import run_path
+parse_deployment_properties = run_path('common.py')['parse_deployment_properties']
+
+
+def upload(url, username, password, local_fn, remote_fn):
+    upload_status_code = sp.check_output([
+        'curl', '--silent', '--output', '/dev/stderr',
+        '--write-out', '%{http_code}',
+        '-u', '{}:{}'.format(username, password),
+        '--upload-file', local_fn,
+        urljoin(url, remote_fn)
+    ]).decode().strip()
+
+    if upload_status_code != '201':
+        raise Exception('upload of {} failed, got HTTP status code {}'.format(
+            local_fn, upload_status_code))
+
+
+def unpack_args(_, a):
+    return a
+
+
+if len(sys.argv) < 2:
+    raise ValueError('Should pass <snapshot|release> as arguments')
+
+repo_type = unpack_args(*sys.argv)
+
+username, password = os.getenv('DEPLOY_DISTRIBUTION_USERNAME'), os.getenv('DEPLOY_DISTRIBUTION_PASSWORD')
+
+if not username:
+    raise ValueError('Error: username should be passed via $DEPLOY_DISTRIBUTION_USERNAME env variable')
+
+if not password:
+    raise ValueError('Error: password should be passed via $DEPLOY_DISTRIBUTION_PASSWORD env variable')
+
+deployment_properties = parse_deployment_properties('deployment.properties')
+distribution_url = deployment_properties['repo.distribution.' + repo_type]
```

Comment:
@adammitchelldev we're accessing `repo_type` before validating its correctness

Reply:
Good spot btw